### ISSUE TITLE
"nadir" fires thousands of times

### ISSUE THREAD

**Issue by Putzeimer** [biddster/node-red-contrib-schedex#57]
I've got 7 nodes with different named on/off times (on: sunrise/off: sunriseEnd; on: dusk/off: dawn; on: solarNoon/off: nadir; etc.). The output is written in a logfile to see, when which event triggers. After one day I got 241839 lines in the file: 1 line for each of the 13 eventnames but 241826 lines for "nadir".
Now, after 4 days, the file contains about 1 million lines. Am I doing something wrong?
The used version is 1.6.2.

Don't need nadir, though. I was just checking how the node works.

![schedex](https://user-images.githubusercontent.com/31502969/68468719-dee14f80-0218-11ea-8c7a-87b595887cad.png)

**Comment by biddster** [comment]
Doesn't look like you're doing anything wrong. I don't know why nadir should be a problem, but I'll drop a schedex  in and try it.

**Comment by biddster** [comment]
Yup - there's a bug here. I'll fix it.

**Comment by biddster** [comment]
Nadir gives yesterday's date, interesting:
```
{
    "solarNoon": "2019-11-17T11:49:40.113Z",
    "nadir": "2019-11-16T23:49:40.113Z",
    "sunrise": "2019-11-17T07:24:47.098Z",
    "sunset": "2019-11-17T16:14:33.128Z",
    "sunriseEnd": "2019-11-17T07:28:44.485Z",
    "sunsetStart": "2019-11-17T16:10:35.741Z",
    "dawn": "2019-11-17T06:47:48.497Z",
    "dusk": "2019-11-17T16:51:31.729Z",
    "nauticalDawn": "2019-11-17T06:07:07.184Z",
    "nauticalDusk": "2019-11-17T17:32:13.042Z",
    "nightEnd": "2019-11-17T05:27:52.163Z",
    "night": "2019-11-17T18:11:28.063Z",
    "goldenHourEnd": "2019-11-17T08:18:28.863Z",
    "goldenHour": "2019-11-17T15:20:51.363Z"
}
```

**Comment by biddster** [comment]
It's a bug in the library I'm using to calculate sun times:

https://github.com/mourner/suncalc/issues/124

**Comment by biddster** [comment]
Fixed in 1.6.4

**Comment by Putzeimer** [comment]
Confirmed that nadir now fires only once per night. But I recognized that it now doesn't fire every night.
I started recording nadir at 5th of january. Here are the days when it successfully fired:
06, 07, 08, 09, 10, 11, 12, 13, 14, (missing), 16, (missing), 18, 19, (missing), 21, 22, 23, 24, 25, 26, (missing), 28, (missing), 30, 31.

**Comment by biddster** [comment]
Thanks for the report. I'll add similar logging to my node-red and see what happens.

**Comment by biddster** [comment]
Yeah, confirmed. I'll work on a fix.

**Comment by biddster** [comment]
Another fix released in 1.10.1

**Comment by Putzeimer** [comment]
29 nadir events in the past 29 nights. None missing, none twice. Fix confirmed. Thanks.
